### ISSUE TITLE
Missing @Override annotation in some generated functional interfaces

### ISSUE THREAD

**Issue by ivaniesta14** [eclipse/xtext-xtend#831]
When Xtend creates a `FunctionalInterface` via a cast (ex. `IntFunction`; see also #830), the functional method is created without the `@Override` annotation, generating a warning.
```xtend
stream.toArray([<Foo>newArrayOfSize(it)] as IntFunction<Foo>)
```
generates
```xtend
final Function1<Integer, Foo[]> _function = (Integer it) -> {
  return new Foo[(it).intValue()];
};
stream.<Foo>toArray(
          ((IntFunction<Foo[]>) new IntFunction<Foo[]>() {
              public Foo[] apply(int value) { //Missing @Override here
                return _function.apply(value);
              }
          };
```
This also happens in the first part of a try-with-resources block:
```xtend
try(val it=new BufferedReader(new FileReader(file))){/*...*/}
```
generates
```xtend
try (final BufferedReader it = new Function0<BufferedReader>() {
        public BufferedReader apply() { //Missing @Override here
          try {
            FileReader _fileReader = new FileReader(file);
            return new BufferedReader(_fileReader);
          } catch (Throwable _e) {
            throw Exceptions.sneakyThrow(_e);
          }
        }
      }.apply()) {/*...*/}
```

**Comment by cdietrich** [comment]
@szarnekow is this a feature we support?

**Comment by cdietrich** [comment]
looks like this happens for all closures that get compiled into an anonymous inner class

org.eclipse.xtext.xbase.compiler.XbaseCompiler.toAnonymousClass(XClosure, ITreeAppendable, LightweightTypeReference, JvmOperation)

**Comment by szarnekow** [comment]
The cast is not supported completely. What does work though, is proper target typing:
```
val IntFunction<..> newArr = [ newArrayOfSize(it) ]
stream.toArray(newArr)
```
or 
```
stream.<Foo>toArray([<Foo>newArrayOfSize(it)])
```

**Comment by szarnekow** [comment]
To me, this is more of a workaround so the bug report is appreciated.

**Comment by cdietrich** [comment]
@szarnekow this bug is about the missing override which will be missing for all closures.

**Comment by cdietrich** [comment]
Fixed in 2.19
